### ISSUE TITLE
cy.intercept not extracting/setting filename from response Content-Disposition header when stubbing a download file call

### ISSUE THREAD

**Issue by FrancoisCamus** [cypress-io/cypress#16365]
I'm migrating cy.route to cy.intercept in a project. I'm unable to get my download file test working with cy.intercept. It works fine with cy.route though.

In my application, I have a download button that, when clicked, calls an api to download a file. Once downloaded, I programmatically populate an associated anchor href and download attributes and clicks the anchor to let the browser prompt to save the file somewhere.

In my test, I'm stubbing the api call to download the file and return a fixture. I'm expecting the right filename coming from the Content-Disposition header in the response.

### Current behavior

- When stubbing with cy.route, the download attribute on the anchor is populated with the filename from the Content-Disposition header.

- When stubbing with cy.intercept, the download attribute on the anchor is set to undefined.

I checked the response from the stubbed call and in both cases the Content-Disposition header is set properly.

### Desired behavior

I'd like cy.intercept to behave the same way as cy.route in this scenario.

### Test code to reproduce

Here is my test:

```typescript
describe('Download tests', () => {

  beforeEach(() => {
    cy.server();

    // Deleting the cypress downloads folder before each run
    const downloadsFolder = Cypress.config('downloadsFolder');
    cy.task('deleteFolder', downloadsFolder);
  });

  it('successfully downloads file', () => {
    // Arrange
    const tableId = 'test_table';
    const tableFileName = `${tableId}.csv`;

    // This setup works properly
    // cy.route({
    //   method: 'GET',
    //   url: `some_url`,
    //   response: `fx:${tableFileName},binary`,
    //   headers: {
    //     'Content-Type': 'text/csv',
    //     'Accept': 'application/octet-stream',
    //     'Content-Disposition': `attachment; filename=${tableFileName}; filename*=UTF-8''${tableFileName}`
    //   }
    // }).as('download');

    // This setup does not work properly
    cy.intercept(
      {
        method: 'GET',
        url: `some_url`,
      },
      {
        fixture: tableFileName,
        headers: {
          'Content-Type': 'text/csv',
          'Accept': 'application/octet-stream',
          'Content-Disposition': `attachment; filename=${tableFileName}; filename*=UTF-8''${tableFileName}`
        }
      }
    ).as('download');

    cy.visit('/');

    // Act
    cy.get('.download-file-button').click();

    cy.wait('@download').then((interception) => {

      // Assert

      // Asserting that the anchor is setup properly
      cy.get('.hidden-download-file-anchor').should('have.attr', 'href');
      cy.get('.hidden-download-file-anchor')
        .should('have.attr', 'download')
        .then((filename) => {
          // With cy.route, the filename is properly populated with the value from the Content-Disposition header
          // With cy.intercept, the filename is undefined
          expect(filename).to.equal(tableFileName);
        });
    });
  });
});
```

### Versions

Cypress 7.2.0

**Comment by jennifer-shehane** [comment]
@FrancoisCamus Have you looked at our file download examples? https://github.com/cypress-io/cypress-example-recipes/tree/master/examples/testing-dom__download/cypress/integration

We'd need to have some HTML to run this test code against otherwise to narrow down what is not working. Can you provide a repo with a failing example against an example app with download?

**Comment by FrancoisCamus** [comment]
This issue has been fixed in 7.3.0. Thanks for the great work!
